src/hooks/useUsersPlan.ts

```diff
@@ -0,0 +1,38 @@
+import { IUser } from 'interfaces/user';
+import { useMemo } from 'react';
+import { useInstanceStatus } from './api/getters/useInstanceStatus/useInstanceStatus';
+
+export interface IUsersPlanOutput {
+    planUsers: IUser[];
+    isBilling: boolean;
+}
+
+export const useUsersPlan = (users: IUser[]): IUsersPlanOutput => {
+    const { instanceStatus, isBilling } = useInstanceStatus();
+
+    const planUsers = useMemo(
+        () => computeUsers(users, isBilling, instanceStatus?.seats),
+        [users, isBilling, instanceStatus?.seats]
+    );
+
+    return {
+        planUsers,
+        isBilling,
+    };
+};
+
+const computeUsers = (
+    users: IUser[],
+    isBilling: boolean,
+    seats: number = 0
+) => {
+    if (!isBilling || !seats) return users;
+
+    users
+        .sort((a, b) => a.createdAt.localeCompare(b.createdAt))
+        .forEach((user, i) => {
+            user.paid = i >= seats;
```

Comment:
Looks like it gives the user a paid = true property if index is greater than or equal to seats. So if the index is above false it's not a paid user (and therefore costs money). But yes, this could be clearer.

Reply:
Refactored:
```JS
const calculatePaidUsers = (
    users: IUser[],
    isBillingUsers: boolean,
    seats: number = 0
) => {
    if (!isBillingUsers || !seats) return users;

    users
        .sort((a, b) => a.createdAt.localeCompare(b.createdAt))
        .forEach((user, index) => {
            user.paid = false;

            // If index is greater or equal to seat, the
            // user isn't paid for and we will add use this
            // to add costs and icons in the userlist
            if (index >= seats) {
                user.paid = true;
            }
        });

    return users;
};
```